### PR TITLE
Exercise 1: Todo-App

### PR DESCRIPTION
**Pull request by eschoenawa**

### CODE REVIEW

**Comment by eschoenawa** [comment]
@roschaefer Erbitten Review.

File: VueJS/src/App.vue
Comment (roschaefer) [inline VueJS/src/App.vue:5]:
```suggestion
      :todoList="todoList"
```

File: VueJS/src/App.vue
Comment (roschaefer) [inline VueJS/src/App.vue:6]:
```suggestion
      @delete-todo="deleteTodo"
```
Use shorthands

File: VueJS/src/App.vue
Comment (roschaefer) [inline VueJS/src/App.vue:51]: Use `map` and replace the whole object: https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Global_Objects/Map

File: VueJS/src/components/AddTodo.vue
Comment (roschaefer) [inline VueJS/src/components/AddTodo.vue:25]: Why do you let me click on the button then? :stuck_out_tongue: you could disable the '"Add" button if the todo is empty.

File: VueJS/src/components/AddTodo.vue
Comment (roschaefer) [inline VueJS/src/components/AddTodo.vue:21]: I think there is a shorthand for this: https://vuejs.org/v2/guide/events.html#Event-Modifiers

File: VueJS/src/components/AddTodo.vue
Comment (roschaefer) [inline VueJS/src/components/AddTodo.vue:27]: State is the root of all evil! Try to avoid things like `this.currentId` whenever you can. Here is a stateless example how to implement a numerical ascending id: https://github.com/m-oehme/Systems-Development-and-Frameworks/pull/2#discussion_r339841486

By the way: A random ID would also do the job and is probably a lot easier.

File: VueJS/src/components/Todo.vue
Comment (roschaefer) [inline VueJS/src/components/Todo.vue:52]:
```js
     if (this.editing) return "edit-background";
     if(!this.todo.done) return "default-background";
     return "done-background";
```

why so complicated?

File: VueJS/src/components/TodoList.vue
Comment (roschaefer) [inline VueJS/src/components/TodoList.vue:19]: It feels odd to me that you want to pass up the `id` and `text` separately to later construct it at the top level component. Why not constructing one object in the `Todo` component and pass up the whole thing? :thinking:

File: VueJS/src/components/Todo.vue
Comment (roschaefer) [inline VueJS/src/components/Todo.vue:31]: I like your way of modeling the `done` attribute on the todo itself.

**Review by roschaefer — APPROVED** [review]
Very straightforward! Good modeling of the `todos`. I also like the styling which reminds me of LaTeX beamer templates. Congratulations for passing the first exam.

straight on

Here are some final suggestions:

Screenshot - 2019-10-29T020345 184

**Comment by roschaefer** [comment]
@eschoenawa could you create the PR withing your repo next time? That way you can merge the PR after it gets approved. The PRs should show up here: https://github.com/morhenny/Systems-Development-and-Frameworks/pulls

I will not merge your PR into my repo because this repo should only contain the exercise description.

Also, Github has a special feature to request a code review from sb: 
Screenshot - 2019-10-29T021300 660

**Comment by roschaefer** [comment]
I will close this PR, OK @eschoenawa ?

Next time please open a PR in your fork between two branches. Request a code review from me there.

You can merge the changes in your own fork. I can't merge your changes into the original repo, only exercise descriptions should go there.

**Pull request closed without merging**
